crypto/rand/randfile.c

```diff
@@ -104,6 +104,12 @@ static __FILE_ptr32 (*const vms_fopen)(const char *, const char *, ...) =
 
 #define RFILE ".rnd"
 
+#ifdef EINTR
+# define INTERRUPTED(in) (ferror(in) && errno == EINTR)
```

Comment:
That depends in when `INTERRUPTED` is used.  Further down, it's used directly after `fread`, so that should correctly catch the `errno` value given by whatever `fread` calls down the line.